contracts/swayswap_abi/src/main.sw

```diff
@@ -20,4 +25,10 @@ abi Exchange {
     fn swap_with_minimum(min: u64, deadline: u64) -> u64;
     /// Swap ETH <-> Tokens and tranfers to sender.
     fn swap_with_maximum(amount: u64, deadline: u64) -> u64;
+    /// Get information on the liquidity pool.
+    fn get_info() -> PoolInfo;
+    /// Get required amount of coins for a swap_with_minimum.
+    fn swap_with_minimum_forward_amount(amount_to_receive: u64) -> u64;
```

Comment:
I see why we need `swap_with_maximum_forward_amount`, but why is this one needed? 
* `swap_with_maximum` says: here's how much I'm forwarding of token `X` and see if you can swap with `amount` of `Y`. So, asking beforehand how much I should forward is reasonable... The app basically should tell me how much I should forward so that the transaction passes.
* `swap_with_minimum` says: use everything I'm forwarding of token `X` to buy token `Y`. However, make sure I get at least `min` amount of `Y`. I suppose there is a need to know what that `min` should be for the transaction to be successful. If that's what `swap_with_minimum_forward_amount` is trying to do (I think that's what the code does), then it should be called `swap_with_minimum_min_value` or something like that.

Reply:
Good catch. I've been having a hard time actually understanding what we are really doing so I was just copy&pasting things around 😄 

How is this?

https://github.com/FuelLabs/swayswap/pull/58/commits/c8227a7f938582c06486e60be4da51f932ef3da9

Reply:
I think this is reasonable :)